src/windows/player.c

```diff
@@ -264,6 +278,63 @@ void window_player_overview_show_group_dropdown(rct_window *w, rct_widget *widge
 	dropdown_set_checked(network_get_group_index(network_get_player_group(player)), true);
 }
 
+void window_player_overview_show_kick_dropdown(rct_window *w, rct_widget *widget)
+{
+	rct_widget *dropdownWidget;
+	int numItems, i;
+	int player = network_get_player_index((uint8)w->number);
+	if (player == -1) {
+		return;
+	}
+
+	dropdownWidget = widget;
+
+	numItems = KickReasonCount;
+	for (i = 0; i < numItems; i++) {
+		gDropdownItemsFormat[i] = KickReasonStringIds[i];
+		gDropdownItemsArgs[i] = 0;
+	}
+	window_dropdown_show_text(
+		w->x + dropdownWidget->left,
+		w->y + dropdownWidget->top,
+		dropdownWidget->bottom - dropdownWidget->top + 1,
+		w->colours[1],
+		0,
+		numItems
+	);
+	gDropdownDefaultIndex = 0;
+	gDropdownHighlightedIndex = 0;
+}
+
+void window_player_overview_kick_player(rct_window* w, int strReasonIndex, const char* reason)
+{
+	uint8 playerid = (uint8)w->number;
+	int player = network_get_player_index(playerid);
+	if (player == -1) {
+		return;
+	}
+	char msg[128];
+	if (strReasonIndex == STR_KICK_REASON_NO_REASON) {
+		format_string(msg, STR_MULTIPLAYER_KICKED_REASON, NULL);
+	}
+	else if (strReasonIndex == STR_KICK_REASON_OTHER_REASON) {
+		if (!reason || strcmp(reason, "") == 0) {
+			format_string(msg, STR_MULTIPLAYER_KICKED_REASON, NULL);
+		}
+		else {
+#ifndef DISABLE_NETWORK
+			snprintf(msg, NETWORK_DISCONNECT_KICK_REASON_MAX_SIZE, reason);
```

Comment:
Never ever use any version of `printf` with a variable format argument. This is an easy attack vector for malicious purposes. In this case you can probably use `strncpy`, that is if you need the `msg` to be a separate pointer at all (I didn't look at the code very closely).